model/map/src/main/java/org/keycloak/models/map/common/locking/ConcurrentMapLocksUtils.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.common.locking;
+
+import org.keycloak.common.util.Time;
+import org.keycloak.models.locking.LockAcquiringTimeoutException;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.util.concurrent.ConcurrentMap;
+ import java.util.function.Supplier;
+
+public class ConcurrentMapLocksUtils {
+
+    public static final String SEPARATOR = ";";
+    private static final String INSTANCE_IDENTIFIER = getKeycloakInstanceIdentifier();
+
+    /**
+     * Repeatedly attempts to put an entry with the key {@code lockName}
+     * to the {@code map}. Succeeds only if there is no entry with the same
+     * key already.
+     * <p />
+     * Execution of this method is time bounded, if this method does not
+     * succeed within {@code timeoutMilliseconds} it gives up and returns
+     * false.
+     * <p />
+     * There is a pause after each unsuccessful attempt equal to
+     * {@code repeatInterval} milliseconds
+     *
+     * @param map Map that will be used for putting the value
+     * @param lockName Name of the entry
+     * @param timeoutMilliseconds time boundaries in milliseconds
+     * @param repeatInterval Number of milliseconds to wait after each
+     *                       unsuccessful attempt
+     *
+     * @throws LockAcquiringTimeoutException the key {@code lockName} was NOT put into the {@code map}
+     *         within time boundaries
+     */
+    public static void repeatPutIfAbsent(Supplier<ConcurrentMap<String, String>> map, String lockName, long timeoutMilliseconds, int repeatInterval) throws LockAcquiringTimeoutException {
+        String currentOwner;
+        long maximumTime = Time.currentTimeMillis() + timeoutMilliseconds;
+        while((currentOwner = map.get().putIfAbsent(lockName, Time.currentTimeMillis() + SEPARATOR + INSTANCE_IDENTIFIER)) != null
+                && Time.currentTimeMillis() < maximumTime) {
+            try {
+                Thread.sleep(repeatInterval);
```

Comment:
Consider using exponential backoff (or a variant thereof) instead. We already have it (see e.g. `Retry.executeWithBackoff`)